sympy/core/function.py

```diff
@@ -3025,7 +3025,26 @@ def expand_power_base(expr, deep=True, force=False):
     >>> expand_power_base((2*y)**(1+z))
     2**(z + 1)*y**(z + 1)
     >>> ((2*y)**(1+z)).expand()
-    2*2**z*y*y**z
+    2*2**z*y**(z + 1)
+
+    The power that is unexpanded can be expanded safely when
+    ``y != 0`` or ``z + 1 != 0``, otherwise if both are 0 but
+    the power was expanded, the whole expression would
+    become 0 instead of ``2*2**z``:
```

Comment:
I don't think this statement is correct. In particular the `z + 1 != 0` condition is not right. We only need to worry about `z` in the case when `y=0` and in that case if `z+1` is nonreal then `z` is also nonreal so both results give nan and we only need to worry about the real case. What is needed is more like that either `1`, `z` and `z+1` are all nonnegative or they are all nonpositive. Since `1` is nonnegative we need `z` and `z+1` to be nonnegative so in this case we need `z >= 0`:
```python
In [20]: e1.subs({y:0, z:-2}, simultaneous=True)
Out[20]: zoo

In [21]: e2.subs({y:0, z:-2}, simultaneous=True)
Out[21]: nan

In [22]: e1.subs({y:0, z:-1}, simultaneous=True)
Out[22]: 1

In [23]: e2.subs({y:0, z:-1}, simultaneous=True)
Out[23]: nan

In [24]: e1.subs({y:0, z:0}, simultaneous=True)
Out[24]: 0

In [25]: e2.subs({y:0, z:0}, simultaneous=True)
Out[25]: 0
```
The above shows that something like the changes you are making here are needed for the expansion to be valid (although force=True could be used to override).

I'm using `simultaneous=True` above because there's also a separate bug that involves intermediate evaluation to zero:
```python
In [26]: e2.subs(y, 0).subs(z, -2)
Out[26]: 0

In [27]: e2.subs(z, -2).subs(y, 0)
Out[27]: zoo
```
It is this bug that you refer to when you say "the expression would become 0". That is really a bug though in my opinion. The immediate target of the work here should be to fix it so that the expansion rules are valid rather than to work around this other bug in evaluation (#17224).

Reply:
> I don't think this statement is correct

I missed that. There was a similar comment about 40 lines below that. I updated it.